src/printer.js

```diff
@@ -4467,17 +4515,23 @@ function exprNeedsASIProtection(node) {
     return false;
   }
 
-  return exprNeedsASIProtection(getLeftSide(node));
+  return path.call(
+    childPath => exprNeedsASIProtection(childPath, options),
+    ...getLeftSidePathName(path, node)
```

Comment:
Can't use `...` because we support Node v4. 

Might need to do something icky like:

```js
path.call.apply(
  path,
  [childPath => exprNeedsASIProtection(childPath, options)].concat(
    getLeftSidePathName(path, node)
  )
);
```

Reply:
Ah, bummer. Thanks for highlighting, will fix